### ISSUE TITLE
BUG: GC.SuppressFinalize within any block (CC0029)

### ISSUE THREAD

**Issue by raphaelm22** [code-cracker/code-cracker#776]
`public override void Dispose()
{
	try
	{
		if (MyConnection != null)
			(MyConnection.Dispose();
	}
	finally
	{
		GC.SuppressFinalize(this);
	}
}
`

When you add a `GC.SuppressFinalize (this)` in a finally block the codecrack suggests the addition of `SuppressFinalize` method. I believe this is a small bug, or not?

![code](https://cloud.githubusercontent.com/assets/11078528/14539430/c012d67a-0256-11e6-867c-765a66d4fe19.png)

Should work with any block too:

This should not raise a diagnostic:

````csharp
public void Dispose()
{
    if (condition)
        GC.SuppressFinalize(this);
}
````

**Comment by vs-krasnov** [comment]
@giggio @raphaelm22 May I work on this issue?

**Comment by giggio** [comment]
@vs-krasnov Go ahead.

**Comment by giggio** [comment]
@vs-krasnov How is this coming? Are you still working on it?

**Comment by vs-krasnov** [comment]
@giggio Will PR fix for this in a day or two - a bit difficult to find free time in the schedule.

**Comment by giggio** [comment]
Closed on #819.
